sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,575 @@
+import org.apache.druid.java.util.http.client.response.InputStreamResponseHandler;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.server.security.AuthorizerMapper;
+import org.apache.druid.sql.calcite.table.RowSignature;
+import org.apache.druid.timeline.DataSegment;
+import org.jboss.netty.handler.codec.http.HttpMethod;
+import org.jboss.netty.handler.codec.http.HttpResponse;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.concurrent.ExecutionException;
+
+public class SystemSchema extends AbstractSchema
+{
+  private static final Logger log = new Logger(SystemSchema.class);
+
+  public static final String NAME = "sys";
+  private static final String SEGMENTS_TABLE = "segments";
+  private static final String SERVERS_TABLE = "servers";
+  private static final String SEGMENT_SERVERS_TABLE = "segment_servers";
+  private static final String TASKS_TABLE = "tasks";
+  private static final int SEGMENT_SERVERS_TABLE_SIZE;
+
+  private static final RowSignature SEGMENTS_SIGNATURE = RowSignature
+      .builder()
+      .add("segment_id", ValueType.STRING)
+      .add("datasource", ValueType.STRING)
+      .add("start", ValueType.STRING)
+      .add("end", ValueType.STRING)
+      .add("size", ValueType.LONG)
+      .add("version", ValueType.STRING)
+      .add("partition_num", ValueType.STRING)
+      .add("num_replicas", ValueType.LONG)
+      .add("num_rows", ValueType.LONG)
+      .add("is_published", ValueType.LONG)
+      .add("is_available", ValueType.LONG)
+      .add("is_realtime", ValueType.LONG)
+      .add("payload", ValueType.STRING)
+      .build();
+
+  private static final RowSignature SERVERS_SIGNATURE = RowSignature
+      .builder()
+      .add("server", ValueType.STRING)
+      .add("scheme", ValueType.STRING)
+      .add("server_type", ValueType.STRING)
+      .add("tier", ValueType.STRING)
+      .add("curr_size", ValueType.LONG)
+      .add("max_size", ValueType.LONG)
+      .build();
+
+  private static final RowSignature SERVERSEGMENTS_SIGNATURE = RowSignature
+      .builder()
+      .add("server", ValueType.STRING)
+      .add("segment_id", ValueType.STRING)
+      .build();
+
+  private static final RowSignature TASKS_SIGNATURE = RowSignature
+      .builder()
+      .add("task_id", ValueType.STRING)
+      .add("type", ValueType.STRING)
+      .add("datasource", ValueType.STRING)
+      .add("created_time", ValueType.STRING)
+      .add("queue_insertion_time", ValueType.STRING)
+      .add("status", ValueType.STRING)
+      .add("runner_status", ValueType.STRING)
+      .add("duration", ValueType.STRING)
+      .add("location", ValueType.STRING)
```

Comment:
similar to the "servers" table, let's add "host" and "plaintextPort" and "tlsPort" as separate columns

Reply:
hmm, for `tasks` table i think a single `location` column seems better than 3 separate columns as that's what is already there in overlord console as well. Do you still prefer separate columns ?

Reply:
I think "location" is fine as a combined "host:port", but it would be useful to have separate host and port columns as well so that they can be referenced in queries individually

Reply:
ok sounds good, will add 3 more columns for host and ports